### PR TITLE
Position absolute fixes

### LINKED ISSUE #27: Fix: Position Absolute on Profie Page
# Value Statement
As a user with a profile, I want to have a clean layout that is responsive with the screen layout so that I can move around the page without any conflicting rulesets.

# Description

Screenshot 2021-05-05 at 10 07 05

# Acceptance Criteria

- `display:grid` should be used.
- grid children should be placed correctly.

# Tasks
- [x] clean up the `position absolute` in the CSS.

# Complexity

Low

### PR DESCRIPTION
**Pull request by hugh-burgess**
fixes #27

### CODE REVIEW

File: client/src/components/profile-page/Profile.css [hunk 1]
```diff
@@ -2,163 +2,179 @@
...
 }
 
 .profile-wrapper {
-  position: relative;
-  height: 80vh;
+  display: flex;
+  flex-direction: column;
+  width: 100vw;
+}
+/*-------*/
```
Comment (FriederikeM) [inline client/src/components/profile-page/Profile.css]: what's this?

File: client/src/components/profile-page/Profile.css [hunk 2]
```diff
@@ -2,163 +2,179 @@
...
   font-family: "Cookie", cursive;
-  position: absolute;
-  left: 237.27px;
-  top: 90px;
   z-index: 1;
   font-size: 36px;
   color: var(--deep-blue);
 }
 
+/*-------*/
```
Comment (FriederikeM) [inline client/src/components/profile-page/Profile.css]: ?

File: client/src/components/profile-page/Profile.css [hunk 3]
```diff
@@ -2,163 +2,179 @@
...
+  color: var(--deep-blue);
+}
+
+.feedback-list {
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 17px;
+  color: var(--deep-blue);
+}
+
+/*-------*/
```
Comment (FriederikeM) [inline client/src/components/profile-page/Profile.css]: ?

**Review by FriederikeM — APPROVED** [review]
looks good mate. Good job on the positioning. Again, maybe more specific commits. There is a lot of new/changed code

File: client/src/components/profile-page/Edit-Profile.css [hunk 4]
```diff
@@ -49,8 +50,8 @@
   border-radius: 15px;
   width: 30px;
   height: 20px;
-  left: 290.067px;
-  bottom: 36.817px;
+  left: 285.833px;
+  top: 478.183px;
```
Comment (marwinburesch) [inline client/src/components/profile-page/Edit-Profile.css]: I removed the `position` property and the button didn't move much, maybe you don't even need it to be positioned that way, which means you wouldn't have to adjust `left` and `top` here pixel by pixel. Also, the fixed sizes are out of place, you should have a better system for giving this button a size. For example the flexbox `middle-flex-wrapper`...

File: client/src/components/profile-page/EditButton.js [hunk 5]
```diff
@@ -2,10 +2,8 @@ import { Link } from "react-router-dom";
...
 export default function EditButton() {
   return (
-    <div className="edit-wrapper">
-      <Link to="/edit-profile">
-        <div className="edit-button">edit</div>
-      </Link>
-    </div>
+    <Link to="/edit-profile">
+      <div className="edit-button">edit</div>
+    </Link>
```
Comment (marwinburesch) [inline client/src/components/profile-page/EditButton.js:7]: So you're adding this as part of the position absolute fix?

File: client/src/components/profile-page/EditContent.js [hunk 6]
```diff
@@ -17,22 +18,43 @@ export default function EditContent() {
...
+          <p className="feedback blur-effect">feedback</p>
+          <div className="feedback-list blur-effect">
+            <li>the goodest boy</li>
+            <li>loves scritches</li>
+            <li>loves walks</li>
+          </div>
+          <Link to="/profile">
+            <div className="save-button">save</div>
+          </Link>
+        </div>
```
Comment (marwinburesch) [inline client/src/components/profile-page/EditContent.js]: This is a lot of markup being added. 😈 
You might want to consider extracting some components here, even if they are exclusive for this page

File: client/src/components/profile-page/MainContent.js [hunk 7]
```diff
@@ -1,32 +1,52 @@
...
+        <img className="profile-bio" src={bioBubble} alt="#" />
+        <div className="middle-flex-feedback-wrapper">
+          <p className="feedback">feedback</p>
+          <div className="feedback-list">
+            <li>the goodest boy</li>
+            <li>loves scritches</li>
+            <li>loves walks</li>
+          </div>
+        </div>
+      </div>
```
Comment (marwinburesch) [inline client/src/components/profile-page/MainContent.js]: Where have I seen this before? 🤔

File: client/src/components/profile-page/Profile.css [hunk 8]
```diff
@@ -2,163 +2,178 @@
...
-.profile-bubble {
-  position: absolute;
-  left: 150px;
-  top: 25.67px;
-  height: 200px;
+.top-icons-and-text {
+  display: flex;
+  flex-direction: column;
+  height: auto;
+  width: auto;
```
Comment (marwinburesch) [inline client/src/components/profile-page/Profile.css]: These should be set to `auto` by default

File: client/src/components/profile-page/Profile.css [hunk 9]
```diff
@@ -2,163 +2,178 @@
...
-  position: absolute;
-  left: 39.79px;
-  top: 265px;
-  z-index: 2;
+.profile-picture {
+  height: 200px;
+}
+
+.profile-bubble {
+  height: 200px;
```
Comment (marwinburesch) [inline client/src/components/profile-page/Profile.css]: Ugh, fixed sizes 😛

File: client/src/components/profile-page/Profile.css [hunk 10]
```diff
@@ -2,163 +2,178 @@
...
 }
 
 .big-bone-image {
   color: var(--grey-shade);
+  width: 100px;
+  height: 100px;
   position: absolute;
-  width: 136px;
-  height: 150px;
+  top: 423px;
```
Comment (marwinburesch) [inline client/src/components/profile-page/Profile.css]: I think positioning this from bottom would make more sense.

**Review by marwinburesch — CHANGES_REQUESTED** [review]
I think it is an excellent idea to start fixing some `position: absolute`! Although there is a lot of other unrelated code in this PR....
 🐜

**Pull request merged**